ts/components/CiePinpad/index.tsx

```diff
@@ -0,0 +1,175 @@
+    justifyContent: "center",
+    color: variables.colorWhite
+  },
+  placeHolderStyle: {
+    height: 4,
+    marginLeft: 2,
+    marginRight: 2,
+    marginTop: 2
+  }
+});
+
+const width = 36;
+const margin = 2;
+const screenWidth = Dimensions.get("window").width;
+/**
+ * A customized CodeInput component.
+ */
+class CiePinpad extends React.PureComponent<Props, State> {
+  private inputs: ReadonlyArray<TextInput>;
+
+  constructor(props: Props) {
+    super(props);
+    this.inputBoxGenerator = this.inputBoxGenerator.bind(this);
+    this.inputs = [];
+    this.state = {
+      pin: new Array(this.props.pinLength).fill("")
+    };
+  }
+
+  private handleOnChangeText = (text: string, index: number) => {
+    const pin = this.updatePin(text, index);
+    this.props.onPinChanged(pin.join(""));
+    // if it is not the last element, change focus on next element
+    // handleOnKeyPress is used to handle the Backspace press
+    if (index + 1 < this.inputs.length) {
+      this.inputs[index + 1].focus();
+      return;
+    }
+  };
+
+  // tslint:disable-next-line: readonly-array
+  private updatePin = (text: string, index: number): string[] => {
+    // tslint:disable-next-line: readonly-array
+    const tempPin = [...this.state.pin];
+    // replace the pin char at the index position
+    tempPin.splice(index, 1, text);
+    const pin: ReadonlyArray<string> = tempPin;
+    this.setState({ pin });
+    return tempPin;
+  };
+
+  private handleOnKeyPress = (
+    nativeEvent: TextInputKeyPressEventData,
+    index: number
+  ) => {
+    if (nativeEvent.key === "Backspace") {
+      // check if a deletion is going. If yes, set the focus to the previous input
+      // it works only if the index is not the first element
+      if (
+        (!this.state.pin[index] || this.state.pin[index].length === 0) &&
+        index > 0
+      ) {
+        this.updatePin("", index - 1);
+        this.inputs[index - 1].focus();
+      }
+      return;
+    }
+  };
+
+  private readonly inputBoxGenerator = (i: number) => {
+    const totalMargins = margin * 2 * (this.props.pinLength - 1);
+    const widthNeeded = width * this.props.pinLength + totalMargins;
+
+    // if we have not enough space to place inputs
+    // compute a new width to fit it
+    // consider margin from both sides too
+
+    const targetDimension =
+      widthNeeded > screenWidth
+        ? (screenWidth - totalMargins) / this.props.pinLength
+        : width;
+
+    return (
+      <View style={{ alignItems: "center" }} key={`input_view-${i}`}>
+        <TextInput
+          ref={c => {
+            // collect all inputs refs
+            if (c !== null && this.inputs.length < this.props.pinLength) {
+              // tslint:disable-next-line: no-object-mutation
+              this.inputs = [...this.inputs, c];
+            }
+          }}
+          style={{
+            width: targetDimension,
+            height: targetDimension,
+            textAlign: "center"
```

Comment:
you could move this and maybe **_fontSize_** (try to increase it. Comparing with this https://projects.invisionapp.com/share/NDUE5R9MRV6#/screens I suggest fontSize=20) too to styles on top of file

Reply:
Done